src/TestableIO.System.IO.Abstractions.TestingHelpers/MockDirectory.cs

```diff
@@ -627,39 +627,39 @@ public override void SetLastWriteTimeUtc(string path, DateTime lastWriteTimeUtc)
         /// <inheritdoc />
         public override IEnumerable<string> EnumerateDirectories(string path)
         {
-            mockFileDataAccessor.PathVerifier.IsLegalAbsoluteOrRelative(path, "path");
-
             return EnumerateDirectories(path, "*");
         }
 
         /// <inheritdoc />
         public override IEnumerable<string> EnumerateDirectories(string path, string searchPattern)
         {
-            mockFileDataAccessor.PathVerifier.IsLegalAbsoluteOrRelative(path, "path");
-
             return EnumerateDirectories(path, searchPattern, SearchOption.TopDirectoryOnly);
         }
 
         /// <inheritdoc />
         public override IEnumerable<string> EnumerateDirectories(string path, string searchPattern, SearchOption searchOption)
         {
             mockFileDataAccessor.PathVerifier.IsLegalAbsoluteOrRelative(path, "path");
+            var originalPath = path;
             path = path.TrimSlashes();
             path = mockFileDataAccessor.Path.GetFullPath(path);
             return GetFilesInternal(mockFileDataAccessor.AllDirectories, path, searchPattern, searchOption)
-                .Where(p => !mockFileDataAccessor.StringOperations.Equals(p, path));
+                .Where(p => !mockFileDataAccessor.StringOperations.Equals(p, path))
+                .Select(p => FixPrefix(p, originalPath));
         }
-
+        
+        private string FixPrefix(string path, string originalPath)
+        {
+            var normalizedOriginalPath = mockFileDataAccessor.Path.GetFullPath(originalPath);
+            return originalPath + path.Substring(normalizedOriginalPath.Length);
```

Comment:
Will you always have a path separator here?
To be sure, you could use `Path.Combine` for concatinating the `originalPath` and the remaining path.

Reply:
I'm pretty new to the domain, but it seems to me that we should always have a path separator here.
The code with the `Path.Combine` would look like this:
```
var pathWithoutOriginalPath = path.Substring(normalizedOriginalPath.Length)
    .TrimStart(mockFileDataAccessor.Path.DirectorySeparatorChar);
return mockFileDataAccessor.Path.Combine(originalPath, pathWithoutOriginalPath);
```
Do you think this would be an improvement?

Reply:
I think, it would be a safer choice. When I look only at the `FixPrefix` method, there is no way to ensure that there is exactly one `DirectorySeparatorChar` either at the end of `originalPath` or at the beginning of `path.Substring(normalizedOriginalPath.Length)`. With your suggestion above, it would make sure, that this is always the case.

Reply:
Alright, changed in 617eca2